File: banking_flutter/lib/src/models/beneficiary.dart
class Beneficiary {
  final String id;
  final String name;
  final String accountNumber;
  final String? bankCode;
  final String? bankName;

  Beneficiary({
    required this.id,
    required this.name,
    required this.accountNumber,
    this.bankCode,
    this.bankName,
  });

  factory Beneficiary.fromJson(Map<String, dynamic> json) {
    return Beneficiary(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      accountNumber: json['accountNumber'] as String? ?? '',
      bankCode: json['bankCode'] as String?,
      bankName: json['bankName'] as String?,
    );
  }
}

File: banking_core_java/src/main/java/com/chidibank/core/domain/Beneficiary.java
package com.chidibank.core.domain;
import lombok.Builder;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Builder
public class Beneficiary {
    private String id;
    private String userId;
    private String accountId;
    private String name;
    private String nickname;
    private String bankCode;
    private String accountNumber;
    private String accountName;
    private String bankName;
    private String relationship;
    private Double dailyLimit;
    private Double monthlyLimit;
    private boolean isActive;
    private boolean isVerified;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
